src/EagerPager.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { findNodeHandle, UIManager } from 'react-native';
+import type { EagerPagerProps } from './types';
+
+import { ViewPagerNative } from './ViewPagerNative';
+
+export class EagerPager extends React.PureComponent<EagerPagerProps> {
```

Comment:
Mimics api of current version of viewpager; render all pages in place (as react children), as opposed to requiring managing a render function (similar to the ScrollView vs FlatList usage).  Component could technically be implemented user side, but is would be more efficient here, as it directly wraps native (instead of wrapping the complications of js-side lazy load).